Do changelog generation in a separate process

mtreinish opened the pull request:
The changelog generation was previously happening synchronously. This
can however take a while because it can involve hundreds of github api
calls to look up each PR that's been merged for a particular release.
The time this takes can exceed the HTTP timeout on the webhook, which
will result in an error on the github side. This isn't critical because
it doesn't stop the bot from running or cause any errors.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Qiskit/qiskit-bot?pullRequest=1) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Qiskit/qiskit-bot?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Qiskit/qiskit-bot?pullRequest=1) it.</sub>

mtreinish committed:
Do changelog generation in a separate process

The changelog generation was previously happening synchronously. This
can however take a while because it can involve hundreds of github api
calls to look up each PR that's been merged for a particular release.
The time this takes can exceed the HTTP timeout on the webhook, which
will result in an error on the github side. This isn't critical because
it doesn't stop the bot from running or cause any errors.
